go-controller/pkg/ovn/gateway_init.go

```diff
@@ -316,12 +316,33 @@ func addDistributedGWPort() error {
 		return fmt.Errorf("failed to get master's chassis ID error: %v", err)
 	}
 
+	var dgpMac string
+	var nbctlArgs []string
 	// add a distributed gateway port to the distributed router
 	dgpName := routerToSwitchPrefix + nodeLocalSwitch
-	dgpIP := net.ParseIP(util.V4NodeLocalDistributedGwPortIP)
-	nbctlArgs := []string{
-		"--may-exist", "lrp-add", ovnClusterRouter, dgpName, util.IPAddrToHWAddr(dgpIP).String(),
-		fmt.Sprintf("%s/%d", dgpIP.String(), util.V4NodeLocalNatSubnetPrefix),
+	if config.IPv4Mode {
+		dgpIP := net.ParseIP(util.V4NodeLocalDistributedGwPortIP)
+		dgpMac = util.IPAddrToHWAddr(dgpIP).String()
+		nbctlArgs = append(nbctlArgs,
+			"--may-exist", "lrp-add", ovnClusterRouter, dgpName, dgpMac,
+			fmt.Sprintf("%s/%d", dgpIP.String(), util.V4NodeLocalNatSubnetPrefix),
+		)
+	}
+	if config.IPv6Mode {
+		dgpIP := net.ParseIP(util.V6NodeLocalDistributedGwPortIP)
+		// If IPv6 Only, mac won't be set. generate mac and add lrp-add cmd.
+		if dgpMac == "" {
+			dgpMac = util.IPAddrToHWAddr(dgpIP).String()
+			nbctlArgs = append(nbctlArgs,
+				"--may-exist", "lrp-add", ovnClusterRouter, dgpName, dgpMac,
+				fmt.Sprintf("%s/%d", dgpIP.String(), util.V6NodeLocalNatSubnetPrefix),
+			)
+		} else {
+			//Dual-stack. Keep same mac and append IPv6 Addess
+			nbctlArgs = append(nbctlArgs,
+				fmt.Sprintf("%s/%d", dgpIP.String(), util.V6NodeLocalNatSubnetPrefix),
```

Comment:
I don't like how this non-obivously assumes what the ipv4 branch left in `nbctlArgs`. It would be better to add the "constant" part of the args first, then have ipv4/ipv6/dual-stack parts to add the ipv4/ipv6/dual-stack addresses